core/src/test/java/org/elasticsearch/gateway/MetaDataWriteDataNodesTests.java

```diff
@@ -45,215 +50,73 @@
-        String redNode = startDataNode("red");
-
-        assertAcked(prepareCreate("blue_index").setSettings(Settings.builder().put("index.number_of_replicas", 0).put(FilterAllocationDecider.INDEX_ROUTING_INCLUDE_GROUP + "color", "blue")));
-        index("blue_index", "doc", "1", jsonBuilder().startObject().field("text", "some text").endObject());
-        assertAcked(prepareCreate("red_index").setSettings(Settings.builder().put("index.number_of_replicas", 0).put(FilterAllocationDecider.INDEX_ROUTING_INCLUDE_GROUP + "color", "red")));
-        index("red_index", "doc", "1", jsonBuilder().startObject().field("text", "some text").endObject());
-        ensureGreen();
-        assertIndexNotInMetaState(blueNode, "red_index");
-        assertIndexNotInMetaState(redNode, "blue_index");
-        assertIndexInMetaState(blueNode, "blue_index");
-        assertIndexInMetaState(redNode, "red_index");
-        assertIndexInMetaState(masterNode, "red_index");
-        assertIndexInMetaState(masterNode, "blue_index");
-
-        // not the index state for blue_index should only be written on blue_node and the for red_index only on red_node
-        // we restart red node and master but with empty data folders
-        stopNode(redNode);
-        ((InternalTestCluster) cluster()).stopCurrentMasterNode();
-        masterNode = startMasterNode();
-        redNode = startDataNode("red");
-
-        ensureGreen();
-        assertIndexNotInMetaState(blueNode, "red_index");
-        assertIndexInMetaState(blueNode, "blue_index");
-        assertIndexNotInMetaState(redNode, "red_index");
-        assertIndexNotInMetaState(redNode, "blue_index");
-        assertIndexNotInMetaState(masterNode, "red_index");
-        assertIndexInMetaState(masterNode, "blue_index");
-        // check that blue index is still there
-        assertFalse(client().admin().indices().prepareExists("red_index").get().isExists());
-        assertTrue(client().prepareGet("blue_index", "doc", "1").get().isExists());
-        // red index should be gone
-        // if the blue node had stored the index state then cluster health would be red and red_index would exist
-        assertFalse(client().admin().indices().prepareExists("red_index").get().isExists());
-
+        assertIndexInMetaState(dataNode, "test");
+        assertIndexInMetaState(masterNode, "test");
     }
 
     @Test
     public void testMetaIsRemovedIfAllShardsFromIndexRemoved() throws Exception {
         // this test checks that the index state is removed from a data only node once all shards have been allocated away from it
-        String masterNode = startMasterNode();
-        String blueNode = startDataNode("blue");
-        String redNode = startDataNode("red");
+        String masterNode = internalCluster().startMasterNode(Settings.EMPTY);
+        List<String> nodeNames = internalCluster().startNodesAsync(2, Settings.builder().put("node.data", true).put("node.master", false).build()).get();
```

Comment:
can we add async variants to do this? startMasterNodesAsync, startDataNodesAsync..